include/openthread/dns_client.h

```diff
@@ -80,6 +80,19 @@ typedef enum
     OT_DNS_NAT64_DISALLOW    = 2, ///< Do not allow NAT64 address translation during DNS client address resolution.
 } otDnsNat64Mode;
 
+/**
+ * This enumeration type represents the DNS transport protocol in an `otDnsQueryConfig`.
+ *
+ * This `OT_DNS_TCP_TRANSPORT` is only used when `OPENTHREAD_CONFIG_DNS_CLIENT_OVER_TCP_ENABLE` is enabled.
+ *
+ */
+typedef enum
+{
+    OT_DNS_UNSPECIFIED_TRANSPORT = 0, /// Dns transport is unspecified.
+    OT_DNS_UDP_TRANSPORT         = 1, /// Dns query should be sent via UDP.
+    OT_DNS_TCP_TRANSPORT         = 2, /// Dns query should be sent via TCP.
```

Comment:
nit: how about using `OT_DNS_TRANSPORT_` prefix?
```suggestion
    OT_DNS_TRANSPORT_UNSPECIFIED = 0, /// DNS transport is unspecified.
    OT_DNS_TRANSPORT_UDP         = 1, /// DNS query should be sent via UDP.
    OT_DNS_TRANSPORT_TCP         = 2, /// DNS query should be sent via TCP.
```

Reply:
fixed